src/kmeans.jl

```diff
@@ -116,35 +146,32 @@ function _kmeans!(
         else
             # if only a small subset is affected, only compute for that subset
             affected_inds = findall(to_update)
-            dmat_p = pairwise(distance, centers[:, affected_inds], X)
-            dmat[affected_inds, :] .= dmat_p
+            pairwise!(view(dmat, affected_inds, :), distance,
```

Comment:
I think it's worth benchmarking this change, as sometimes it can be more efficient to make a copy (since computing distances requires going over the array many times).

Reply:
see small benchmarks at the top, this definitely made a positive change towards fewer allocations and slightly faster overall.

Reply:
actually I tried changing just that back to what it was:

```julia
126.114 ms (280 allocations: 1.63 MiB)
75.811 ms (183 allocations: 1.13 MiB)
```

so it is worse in terms of allocations (timings are not useful here bc I'm another computer).